docs/Projects/TAD/ADRs/0004-software-stack.md

```diff
@@ -0,0 +1,78 @@
+# ADR-0004 Software Stack for TAD
+
+## Context
+
+For building our own TAD solution, we need to choose a software stack. During our earlier
+POCs and market research, we gathered insight and information on technologies to use
+and which not to use.
+
+During further discussions and brainstorm sessions, a software stack was chosen
+that accommodates our needs best.
+
+While more fine grained [requirements](../Existing-Tools/Comparison/requirements) are listed elsewhere,
+some key requirements are:
+
+* the tool is runnable on local dev machines
+* the tool is runnable as user as a local Docker solution
+* the tool is runnable on a cloud platform based on Kubernetes
+* the tool supports multiple organizations, teams and projects
+* the tool supports oath / keycloak
+
+## Assumptions
+
+We stick to one programming language to avoid unneeded complexity. As most AI related
+tooling is written in Python, this is the logical choice for our development as well.
+
+There is no need for a complex or separate web GUI framework. Also to avoid more components and complexity,
+it is preferred to bundle backend and frontend in one solution.
+
+As a government organization, we need to adhere to the digital inclusion standards, like
+[Digitoegankelijk](https://www.digitoegankelijk.nl/) and
+[WCAG Guidelines](https://wcag.nl/kennis/richtlijnen/wcag-2-1-richtlijnen/) .
+
+We do not forsee high load or many static files, therefor a separate webserver like
+nginx, apache is not required. Static files will be served directly by the backend.
+
+## Decision
+
+### Programming language
+
+We will use [Python](https://www.python.org/) v3.11.X as programming language
+and [Poetry](https://python-poetry.org/) for dependency management.
+
+### Backend
+
+The Python backend will use the following key dependencies:
+
+* [Granian](https://github.com/emmett-framework/granian) as HTTP server
+* [Pydantic](https://docs.pydantic.dev/latest/) for data validation
+* [FastAPI](https://fastapi.tiangolo.com/) for REST/API/HTML communication
+* [Jinja2](https://palletsprojects.com/p/jinja/) for templating
```

Comment:
```suggestion
* [Granian](https://github.com/emmett-framework/granian) as HTTP server.
* [Pydantic](https://docs.pydantic.dev/latest/) for data validation.
* [FastAPI](https://fastapi.tiangolo.com/) for REST/API/HTML communication.
* [Jinja2](https://palletsprojects.com/p/jinja/) for templating.
```

Reply:
Thanks, solved this in another commit to avoid conflicts.